pkg/asset/store.go

```diff
@@ -53,25 +51,58 @@ func NewStore(dir string) (Store, error) {
 		assets:      make(map[reflect.Type]assetState),
 	}
 
-	if err := store.load(dir); err != nil {
+	if err := store.load(); err != nil {
 		return nil, err
 	}
-
 	return store, nil
 }
 
 // Fetch retrieves the state of the given asset, generating it and its
 // dependencies if necessary.
 func (s *StoreImpl) Fetch(asset Asset) error {
-	_, err := s.fetch(asset, "")
-	return err
+	if _, err := s.fetch(asset, ""); err != nil {
+		return err
+	}
+	if err := s.save(); err != nil {
+		return errors.Wrapf(err, "failed to save state")
+	}
+	if wa, ok := asset.(WritableAsset); ok {
+		return errors.Wrapf(s.purge([]WritableAsset{wa}), "failed to purge asset")
+	}
+	return nil
+}
+
+// Destroy removes the asset from all its internal state and also from
+// disk if possible.
+func (s *StoreImpl) Destroy(asset Asset) error {
+	if sa, ok := s.assets[reflect.TypeOf(asset)]; ok {
+		reflect.ValueOf(asset).Elem().Set(reflect.ValueOf(sa.asset).Elem())
+	} else if s.isAssetInState(asset) {
+		if err := s.loadAssetFromState(asset); err != nil {
+			return err
+		}
+	} else {
+		// nothing to do
+		return nil
+	}
+
+	if wa, ok := asset.(WritableAsset); ok {
+		logrus.Debugf("Purging asset %q from disk", wa.Name())
```

Comment:
Push this log down into `DeleteAssetFromDisk`?  It seems important enough to get a debug log whenever it happens, not just when it's called from `Destroy`.